modules/video_input/video_input.py

```diff
@@ -3,23 +3,72 @@
 """
 
 from .. import image_and_time
-from ..common.modules.camera import camera_device
+from ..common.modules.camera import base_camera
+from ..common.modules.camera import camera_factory
+from ..common.modules.camera import camera_opencv
+from ..common.modules.camera import camera_picamera2
+from ..common.modules.logger import logger
 
 
 class VideoInput:
     """
     Combines image and timestamp together.
     """
 
-    def __init__(self, camera_name: "int | str", save_name: str = "") -> None:
-        self.device = camera_device.CameraDevice(camera_name, 1, save_name)
+    __create_key = object()
 
-    def run(self) -> "tuple[bool, image_and_time.ImageAndTime | None]":
+    @classmethod
+    def create(
+        cls,
+        camera_option: camera_factory.CameraOption,
+        width: int,
+        height: int,
+        config: camera_opencv.ConfigOpenCV | camera_picamera2.ConfigPiCamera2,
+        save_prefix: str,
```

Comment:
Rename to `maybe_image_name` with type `str | None` . Then images are only logged if this variable is not none, and an empty string is valid input for logging.